per/drf_views.py

```diff
@@ -362,3 +365,19 @@ def get_queryset(self):
                 "-assessment_number").distinct("country_id").values("id")
         )
         return self.get_filtered_queryset(self.request, queryset, dispatch=0)
+
+
+class OpsLearningViewset(viewsets.ModelViewSet):
+    """
+    A simple ViewSet for viewing and editing OpsLearning records.
+    """
+    queryset = OpsLearning.objects.all()
+    serializer_class = OpsLearningSerializer
+    permission_classes = [IsAuthenticated, OpsLearningPermission]
+
+    def get_serializer_class(self):
+        user = self.request.user
+        if not user.is_superuser:
+            return OpsLearningSerializer
+        else:
+            return PublicOpsLearningSerializer
```

Comment:
I am a bit confused here, is this logic for superuser or just logged-in? Maybe something like this?
```suggestion
        if not user.is_authenticated:
            return PublicOpsLearningSerializer
        return  OpsLearningSerializer
```

Reply:
You were correct, it was erroneous: 'if **not** user.is_superuser'.
I fixed the mistakes and tried to implement https://github.com/IFRCGo/go-api/issues/1927#issuecomment-1840510193 permission requirements.